packages/create-sitecore-jss/src/initializers/nextjs-sxa/index.ts

```diff
@@ -28,6 +29,13 @@ export default class NextjsSxaInitializer implements Initializer {
 
     await transform(templatePath, mergedArgs);
 
+    if (
+      !args.templates.includes('nextjs-styleguide') &&
```

Comment:
The problem which you need to solve is that if nextjs-styleguide and nextjs-sxa are installed together - you need to show an incompatibility message, instead of a missing addon message. But you are checking if it's missed.
And additionally, you need to show a warning message when you select add-ons, even before accepting, and logic happens here: https://github.com/Sitecore/jss/blob/dev/packages/create-sitecore-jss/src/initializers/nextjs/prompts.ts#L51